app/kuma-cp/cmd/run.go

```diff
@@ -96,6 +96,10 @@ func newRunCmdWithOpts(opts runCmdOpts) *cobra.Command {
 					runLog.Error(err, "unable to set up GC")
 					return err
 				}
+				if err := rt.Add(rt.DpServer()); err != nil {
```

Comment:
I started with
```
func SetupServer(rt Runtime) error {
  rt.Add(rt.DpServer())
}
```
but then it introduced cyclic dependency. I can however put this file into subpackage `components`. WDYT?

Reply:
What is the reason for cyclic dependency? Most of the components (except `dns`) have `SetupServer` func right in the component's package

Reply:
Runtime depends on Dp Server (because we are using it later to register APIs, so we need it in Runtime) and Dp Server depends on Runtime.

Reply:
I think `DpServer` as a component should implement only `Start` and `NeedLeaderElection`. 
In the method `dp_server.SetupServer` we should register xds and sds:
```go
func SetupServer(rt Runtime) error {
  xdsServer := ...
  sdsServer := 
  dpServer := NewDpServer(xdsServer, sdsServer)
  return rt.Add(dpServer)
}
```
I think it makes more sense because you see what dpServer consists of. You don't need to look through all usages of `RegisterHTTPEndpoint` to figure it out.